cmd/maya-apiserver/app/server/volume_endpoint.go

```diff
@@ -318,3 +322,19 @@ func (s *HTTPServer) volumeAdd(resp http.ResponseWriter, req *http.Request) (int
 
 	return details, nil
 }
+
+// getVolumeCloneDetails gets source volume controller-ip and storage class
+// information for creating clone volume for clone enabled volume.
+func (s *HTTPServer) getVolumeCloneDetails(resp http.ResponseWriter, req *http.Request, vol *v1.Volume) error {
+	glog.Infof("Processing Volume clone for volume : '%s', request for volume '%s'", vol.Name, vol.SourceVolume)
+
+	voldetails, err := s.volumeRead(resp, req, vol.SourceVolume)
+	if err != nil {
+		return err
+	}
+	vol.CloneIP = voldetails.Annotations[string(v1.ControllerIPsAPILbl)]
+	if len(vol.Labels.K8sStorageClass) == 0 {
```

Comment:
When will the storage class be already set. Can we put some code level comments.

Reply:
Added the required comment, thanks.